src/metabase/queries/models/card.clj

```diff
@@ -670,6 +672,26 @@
   (update card :result_metadata (fn [cols]
                                   (mapv #(dissoc % :ident :model/inner_ident) cols))))
 
+;; Schema upgrade: 22 to 23 ==========================================================================================
+;; #66199 shows a customer upgrading to 57 breaking certain cards
+;; The cards in question have {} as their dataset_query.
+;; This schema upgrade is to overwrite cards with dataset_query = "{}" with a converted legacy_query.
+;; Discussion: https://metaboat.slack.com/archives/C05MPF0TM3L/p1764944945966649?thread_ts=1763990244.080579&cid=C05MPF0TM3L
+(defmethod upgrade-card-schema-to 23
+  [card _schema-version]
+  (cond
+    (seq (:dataset_query card))
+    card
+
+    (not (seq (:legacy_query card)))
+    card
+
+    :else
+    (assoc card :dataset_query (-> (:legacy_query card)
+                                   (mi/json-out-without-keywordization)
+                                   (leg-normalize/normalize-or-throw)
+                                   (lib.convert/->pMBQL)))))
```

Comment:
Per style guide, no parens on unary functions in threading macros.

Reply:
Also I'm not sure about throwing here. That will block reading of broken cards altogether. Though perhaps trying to work with cards with `:dataset_query {}` is worse...